### PR TITLE
Upgrade cookbook and add bionic compatibility

### PR DESCRIPTION
**Pull request by iamjohnnym**
### Description

* Added Ubuntu 18.04 support
* Removed
  * docker
  * java
  * dnsmasq-local
  * snoopy
* Uses Socrata's `kitchen-microwave` framework
* Fixed unit and integration tests

`Docker` was removed since its not in use and `dnsmasq-local` is not
compatible on 18.04.
`dnsmasq-local` isn't currently needed with docker being removed and
lack of compatibility on 18.04.
`snoopy` was removed in favor of the system installed logging packages
`java` was removed in favor of the system level java install

### Issues Resolved

* Integration and spec tests pass
* Bionic 18.04 support

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] CHANGELOG has been updated
- [ ] Version has been bumped in metadata
- [x] Commits have been squashed, as appropriate
- [x] Commits include descriptive messages, subjects written in the imperative

### CODE REVIEW

File: recipes/remote_desktop.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@
 # limitations under the License.
 #
 
+package('gpg-agent') if node['platform_version'] == '18.04'
```
Comment (znep) [inline recipes/remote_desktop.rb]: what requires gpg-agent and why only 18.04?

Comment (iamjohnnym) [reply recipes/remote_desktop.rb]: This has been something I've noticed in any cookbook where we are adding a repository and add key from the keyserver.  On 18.04, it doesn't appear to have any sort of `gpg` packages installed...

Actually, I should back that up.  I have noticed this being a thing specifically on chef-dokken images; I am unsure about AWS AMI's but maybe.

**Comment by iamjohnnym** [comment]
I haven't tested the X2Go.  I'm giving that a go this morning since I finally got the cookbook to a place as of Friday.

File: recipes/remote_desktop.rb (hunk 1 above)
Comment (iamjohnnym) [reply recipes/remote_desktop.rb]: Removed this.  Installing `gnugp` by default instead.

**Comment by iamjohnnym** [comment]
I've successfully tested x2go with kitchen-dokken.

**Pull request merged**